From b0691826c330b4b0c148404936a1af26d9fabdf0 Mon Sep 17 00:00:00 2001
From: "T R, Thejesh Reddy" <thejesh.reddy.t.r@intel.com>
Date: Wed, 2 Mar 2022 19:31:45 +0530
Subject: [PATCH] Passing dwc instead of dwc->base as per read/write function
 definition

As per definition of dwc_pwm_readl and writel functions,dwc structure is to
be passed rather than dwc->base

Signed-off-by: T R, Thejesh Reddy <thejesh.reddy.t.r@intel.com>
---
 drivers/pwm/pwm-dwc.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/pwm/pwm-dwc.c b/drivers/pwm/pwm-dwc.c
index 25ec90adc097..081f06f7cc71 100644
--- a/drivers/pwm/pwm-dwc.c
+++ b/drivers/pwm/pwm-dwc.c
@@ -312,20 +312,20 @@ static int dwc_pwm_runtime_suspend(struct device *dev)
 	j0 = jiffies;
 	j1 = j0 + delay;
 
-	cgsr_reg = dwc_pwm_readl(dwc->base, DWC_TIM_CGSR);
-	dwc_pwm_writel(dwc->base, DWC_TIM_CGSR, DWC_TIM_CGSR_CG);
+	cgsr_reg = dwc_pwm_readl(dwc, DWC_TIM_CGSR);
+	dwc_pwm_writel(dwc, DWC_TIM_CGSR, DWC_TIM_CGSR_CG);
 
-	d0i3c_reg = dwc_pwm_readl(dwc->base, DWC_TIM_D0I3C);
+	d0i3c_reg = dwc_pwm_readl(dwc, DWC_TIM_D0I3C);
 
 	if (d0i3c_reg & DWC_TIM_D0I3_CIP) {
 		dev_info(dev, "%s d0i3c CIP detected", __func__);
 	} else {
-		dwc_pwm_writel(dwc->base, DWC_TIM_D0I3C, DWC_TIM_D0I3_EN);
-		d0i3c_reg = dwc_pwm_readl(dwc->base, DWC_TIM_D0I3C);
+		dwc_pwm_writel(dwc, DWC_TIM_D0I3C, DWC_TIM_D0I3_EN);
+		d0i3c_reg = dwc_pwm_readl(dwc, DWC_TIM_D0I3C);
 	}
 
 	while (time_before(jiffies, j1)) {
-		d0i3c_reg = dwc_pwm_readl(dwc->base, DWC_TIM_D0I3C);
+		d0i3c_reg = dwc_pwm_readl(dwc, DWC_TIM_D0I3C);
 		if (!(d0i3c_reg & DWC_TIM_D0I3_CIP))
 			break;
 	}
@@ -347,12 +347,12 @@ static int dwc_pwm_runtime_resume(struct device *dev)
 	u32 d0i3c_reg;
 	u32 cgsr_reg;
 
-	cgsr_reg = dwc_pwm_readl(dwc->base, DWC_TIM_CGSR);
+	cgsr_reg = dwc_pwm_readl(dwc, DWC_TIM_CGSR);
 
 	if (cgsr_reg & DWC_TIM_CGSR_CG)
-		dwc_pwm_writel(dwc->base, DWC_TIM_CGSR, (cgsr_reg & ~DWC_TIM_CGSR_CG));
+		dwc_pwm_writel(dwc, DWC_TIM_CGSR, (cgsr_reg & ~DWC_TIM_CGSR_CG));
 
-	d0i3c_reg = dwc_pwm_readl(dwc->base, DWC_TIM_D0I3C);
+	d0i3c_reg = dwc_pwm_readl(dwc, DWC_TIM_D0I3C);
 
 	if (d0i3c_reg & DWC_TIM_D0I3_CIP) {
 		dev_info(dev, "%s d0i3c CIP detected", __func__);
@@ -363,7 +363,7 @@ static int dwc_pwm_runtime_resume(struct device *dev)
 		if (d0i3c_reg & DWC_TIM_D0I3_RR)
 			d0i3c_reg |= DWC_TIM_D0I3_RR;
 
-		dwc_pwm_writel(dwc->base, DWC_TIM_D0I3C, d0i3c_reg);
+		dwc_pwm_writel(dwc, DWC_TIM_D0I3C, d0i3c_reg);
 	}
 
 	return 0;
-- 
2.17.1

